WordPress/Classes/ViewRelated/Stats/Today Widgets/iOS 14 Widgets/HomeWidgetTodayData.swift

```diff
@@ -0,0 +1,79 @@
+import WidgetKit
+
+// TODO - TODAYWIDGET: we might change this and use only one type for all three widgets
+/// protocol that generalizes home widgets data
+protocol HomeWidgetData: Codable, TimelineEntry {
+
+    associatedtype WidgetStats
+
+    var siteID: Int { get }
+    var siteName: String { get }
+    var url: String { get }
+    var timeZoneName: String { get }
+    var date: Date { get }
+    var stats: WidgetStats { get }
+}
+
+
+// TODO - TODAYWIDGET: we might change this and use only one type for all three widgets
+/// Type that contains all the relevant data of a Today Home Wifget
+struct HomeWidgetTodayData: HomeWidgetData {
+
+    typealias WidgetStats = TodayWidgetStats
+
+    let siteID: Int
+    let siteName: String
+    let url: String
+    let timeZoneName: String
+    let date: Date
+    let stats: WidgetStats
+}
+
+
+// MARK: - Local cache
+extension HomeWidgetTodayData {
+
+    static func read(from cache: HomeWidgetCache<Self>? = nil) -> [Int: HomeWidgetTodayData]? {
+
+        let cache = cache ?? HomeWidgetCache<HomeWidgetTodayData>(fileName: Constants.fileName,
```

Comment:
This is the instantiation that happens at each call, that I referenced in a previous comment.  Just pointing it out so that my previous comment is a bit clearer.

Having to instantiate this each time made me think that maybe `HomeWidgetTodayData` can be instantiable too, and hold one instance of this object during its lifetime.